activerecord/lib/active_record/railtie.rb

```diff
@@ -178,6 +182,11 @@ class Railtie < Rails::Railtie # :nodoc:
     initializer "active_record.check_represent_sqlite3_boolean_as_integer" do
       config.after_initialize do
         ActiveSupport.on_load(:active_record_sqlite3adapter) do
+          sqlite3_represent_boolean_as_integer = Rails.application.config.active_record.delete(:sqlite3_represent_boolean_as_integer)
+          unless sqlite3_represent_boolean_as_integer.nil?
```

Comment:
No lets not go there in this PR. I'd be happy to take a look at an issue that showcased every instance of this pattern and why we have to do it in those cases. Then we've hopefully got a good enough understanding of the cases to figure out where to go next 😊